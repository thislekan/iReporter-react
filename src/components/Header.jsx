import React from 'react';
import { NavLink, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import cx from 'classnames';
import style from '../styles/header.css';

function Header(props) {
  const { location } = props;
  return (
    <div>
      {(location.pathname === '/login' || location.pathname === '/signup') ? '' : <header>
        <nav>
          <div className={style.navigationWrapper}>
            <div
              style={
                {
                  marginLeft: '10px', display: 'flex', alignItems: 'center', justifyContent: 'center',
                }
              }
            >
              <NavLink to='/' exact={true}
                className={cx(style.ireporterLogo, style.navigationLinks)}>iReporter</NavLink>
            </div>
            {(location.pathname === '/admin' || location.pathname === '/user') ? null : <div className={style['login-link']}>
              <NavLink to='/login' className={style.navigationLinks}>Login</NavLink>
            </div>}
            {(location.pathname === '/user' || location.pathname === '/admin') ? <div>
              <NavLink
                onClick={() => localStorage.removeItem('persist:root')}
                to='/'
                className={style['log-out-btn']}
              >
                Log Out
              </NavLink>
            </div> : null}
          </div>
        </nav>
      </header>}
    </div>
  );
}

Header.propTypes = {
  location: PropTypes.object.isRequired,
};

export default withRouter(Header);
