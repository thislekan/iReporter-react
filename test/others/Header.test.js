import React from 'react';
import setup from '../mock/setup';
import Header from '../../src/components/Header.jsx';

describe('<Header />', () => {
  it('should render the header component', () => {
    const wrapper = setup(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render a different content in the header component', () => {
    const wrapper = setup(<Header location={{ pathname: '/signup' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
