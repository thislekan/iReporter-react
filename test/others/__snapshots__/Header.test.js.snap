// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render a different content in the header component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Header)
        location={
          Object {
            "pathname": "/signup",
          }
        }
      >
        <Route>
          <Header
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div>
              <header>
                <nav>
                  <div>
                    <div
                      style={
                        Object {
                          "alignItems": "center",
                          "display": "flex",
                          "justifyContent": "center",
                          "marginLeft": "10px",
                        }
                      }
                    >
                      <NavLink
                        className=""
                        exact={true}
                        to="/"
                      >
                        <Route
                          exact={true}
                          path="\\\\/"
                        >
                          <Link
                            aria-current="page"
                            className="active"
                            style={Object {}}
                            to="/"
                          >
                            <a
                              aria-current="page"
                              className="active"
                              href="/"
                              onClick={[Function]}
                              style={Object {}}
                            >
                              iReporter
                            </a>
                          </Link>
                        </Route>
                      </NavLink>
                    </div>
                    <div>
                      <NavLink
                        to="/login"
                      >
                        <Route
                          path="\\\\/login"
                        >
                          <Link
                            aria-current={null}
                            to="/login"
                          >
                            <a
                              aria-current={null}
                              href="/login"
                              onClick={[Function]}
                            >
                              Login
                            </a>
                          </Link>
                        </Route>
                      </NavLink>
                    </div>
                  </div>
                </nav>
              </header>
            </div>
          </Header>
        </Route>
      </withRouter(Header)>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`<Header /> should render the header component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Header)>
        <Route>
          <Header
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div>
              <header>
                <nav>
                  <div>
                    <div
                      style={
                        Object {
                          "alignItems": "center",
                          "display": "flex",
                          "justifyContent": "center",
                          "marginLeft": "10px",
                        }
                      }
                    >
                      <NavLink
                        className=""
                        exact={true}
                        to="/"
                      >
                        <Route
                          exact={true}
                          path="\\\\/"
                        >
                          <Link
                            aria-current="page"
                            className="active"
                            style={Object {}}
                            to="/"
                          >
                            <a
                              aria-current="page"
                              className="active"
                              href="/"
                              onClick={[Function]}
                              style={Object {}}
                            >
                              iReporter
                            </a>
                          </Link>
                        </Route>
                      </NavLink>
                    </div>
                    <div>
                      <NavLink
                        to="/login"
                      >
                        <Route
                          path="\\\\/login"
                        >
                          <Link
                            aria-current={null}
                            to="/login"
                          >
                            <a
                              aria-current={null}
                              href="/login"
                              onClick={[Function]}
                            >
                              Login
                            </a>
                          </Link>
                        </Route>
                      </NavLink>
                    </div>
                  </div>
                </nav>
              </header>
            </div>
          </Header>
        </Route>
      </withRouter(Header)>
    </Router>
  </BrowserRouter>
</Provider>
`;
